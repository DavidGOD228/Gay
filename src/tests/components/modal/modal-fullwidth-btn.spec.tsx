import 'react-native'
// Note: test renderer must be required after react-native.
import React from 'react'
import renderer from 'react-test-renderer'
import ModalFullwidthBtn from '../../../components/modal/modal-fullwidth-btn'
import {
  Text
} from 'react-native'

it('renders correctly disabled empty', () => {
  const tree = renderer.create(
    <ModalFullwidthBtn onPress={() => null} disabled={true} />
  );
  expect(tree).toMatchSnapshot();
})

it('renders correctly enabled empty', () => {
  const tree = renderer.create(
    <ModalFullwidthBtn onPress={() => null} disabled={false} />
  );
  expect(tree).toMatchSnapshot();
})

it('renders correctly disabled text', () => {
  const tree = renderer.create(
    <ModalFullwidthBtn onPress={() => null} disabled={true}>
      <Text>
        All you had to do was follow the damn train CJ
      </Text>
    </ModalFullwidthBtn>
  );
  expect(tree).toMatchSnapshot();
})

it('renders correctly enabled text', () => {
  const tree = renderer.create(
    <ModalFullwidthBtn onPress={() => null} disabled={false}>
      <Text>
        All you had to do was follow the damn train CJ
      </Text>
    </ModalFullwidthBtn>
  );
  expect(tree).toMatchSnapshot();
})
