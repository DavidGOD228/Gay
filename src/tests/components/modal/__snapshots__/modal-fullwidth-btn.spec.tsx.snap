// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly disabled empty 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#5BDAFD",
      "height": 50,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 22,
              "fontWeight": "bold",
            },
            Object {
              "color": "#ABABAB",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`renders correctly disabled text 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#5BDAFD",
      "height": 50,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 22,
              "fontWeight": "bold",
            },
            Object {
              "color": "#ABABAB",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        >
          All you had to do was follow the damn train CJ
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly enabled empty 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#5BDAFD",
      "height": 50,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 22,
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`renders correctly enabled text 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#5BDAFD",
      "height": 50,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 22,
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        >
          All you had to do was follow the damn train CJ
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
